src/symmetric_crypto/key.rs

```diff
@@ -1,101 +1,89 @@
-        self.0.clone()
+    /// Convert the given key into bytes.
+    #[inline]
+    fn to_bytes(&self) -> [u8; LENGTH] {
+        self.0.to_owned()
     }
 
     /// Try to convert the given bytes into a key. Size must be correct.
+    #[inline]
     fn try_from_bytes(bytes: &[u8]) -> Result<Self, CryptoCoreError> {
-        Self::try_from(bytes)
+        let bytes = <[u8; LENGTH]>::try_from(bytes)
+            .map_err(|e| CryptoCoreError::ConversionError(e.to_string()))?;
+        Ok(Self(bytes))
     }
 }
 
-impl<KeyLength: Eq + ArrayLength<u8>> SymKey for Key<KeyLength> {
-    /// Convert the given key into a byte slice, without copy.
+impl<const LENGTH: usize> SymKey<LENGTH> for Key<LENGTH> {
+    /// Convert the given key into a byte slice.
+    #[inline]
     fn as_bytes(&self) -> &[u8] {
         &self.0
     }
 
-    fn from_bytes(bytes: GenericArray<u8, Self::Length>) -> Self {
-        Self(bytes)
+    /// Consume the key to return underlying bytes.
+    #[inline]
+    fn into_bytes(self) -> [u8; LENGTH] {
+        self.0
     }
-}
-
-impl<KeyLength: Eq + ArrayLength<u8>> From<Key<KeyLength>> for GenericArray<u8, KeyLength> {
-    fn from(k: Key<KeyLength>) -> Self {
-        k.to_bytes()
-    }
-}
 
-impl<KeyLength: ArrayLength<u8>> From<GenericArray<u8, KeyLength>> for Key<KeyLength> {
-    fn from(b: GenericArray<u8, KeyLength>) -> Self {
-        Self(b)
+    /// Convert the given bytes with correct size into a key.
+    #[inline]
+    fn from_bytes(bytes: [u8; LENGTH]) -> Self {
+        Self(bytes)
     }
 }
 
-impl<'a, KeyLength: ArrayLength<u8>> TryFrom<&'a [u8]> for Key<KeyLength> {
-    type Error = CryptoCoreError;
-
-    fn try_from(bytes: &'a [u8]) -> Result<Self, Self::Error> {
-        if bytes.len() != KeyLength::to_usize() {
-            return Err(Self::Error::SizeError {
-                given: bytes.len(),
-                expected: KeyLength::to_usize(),
-            });
-        }
-        Ok(Self(GenericArray::<u8, KeyLength>::clone_from_slice(bytes)))
+impl<const KEY_LENGTH: usize> Display for Key<KEY_LENGTH> {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        write!(f, "{}", hex::encode(self.as_bytes()))
     }
 }
 
-impl<KeyLength: ArrayLength<u8>> Display for Key<KeyLength> {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(f, "{}", hex::encode(self.as_slice()))
+impl<const KEY_LENGTH: usize> Default for Key<KEY_LENGTH> {
+    fn default() -> Self {
+        Self([0; KEY_LENGTH])
     }
 }
 
-impl<KeyLength: ArrayLength<u8>> Zeroize for Key<KeyLength> {
+impl<const KEY_LENGTH: usize> Zeroize for Key<KEY_LENGTH> {
     fn zeroize(&mut self) {
         self.0.zeroize();
     }
 }
 
 // Implement `Drop` trait to follow R23.
-impl<KeyLength: ArrayLength<u8>> Drop for Key<KeyLength> {
+impl<const KEY_LENGTH: usize> Drop for Key<KEY_LENGTH> {
     fn drop(&mut self) {
         self.zeroize();
     }
 }
```

Comment:
Can you use the derive trait of `Zeroize` and `ZeroizeOnDrop` for this simple struct?